### ISSUE TITLE
[bug] `LFS_MISSING_FILES` not printed after #2347

### ISSUE THREAD

**Issue by junliume** [ROCmSoftwarePlatform/MIOpen#2709]
@apwojcik there is another issue on `LFS_MISSING_FILES` from
https://github.com/ROCm/MIOpen/blob/06b2bd395f2629b1542d5b2a5d234195f9b41bb8/CMakeLists.txt#L515

This variable gets cleared from scratch every time the function `install_kdb` is called. I guess we need to make it a global variable somehow?

same issue has been discussed here: https://stackoverflow.com/questions/10031953/how-to-set-the-global-variable-in-a-function-for-cmake

_Originally posted by @junliume in https://github.com/ROCm/MIOpen/issues/2702#issuecomment-1915834498_

**[Observations]:**
When a repo is freshly pulled without `LFS` files been updated:
```
build git: ✗ ls -ltrah ../src/kernels/*.kdb.bz2
-rw-rw-r-- 1 user user 134 Dec 14 16:22 ../src/kernels/gfx1030.kdb.bz2
-rw-rw-r-- 1 user user 134 Dec 14 16:22 ../src/kernels/gfx900.kdb.bz2
-rw-rw-r-- 1 user user 134 Dec 14 16:22 ../src/kernels/gfx906.kdb.bz2
-rw-rw-r-- 1 user user 134 Dec 14 16:37 ../src/kernels/gfx908.kdb.bz2
-rw-rw-r-- 1 user user 134 Dec 14 16:37 ../src/kernels/gfx90a.kdb.bz2
-rw-rw-r-- 1 user user 132 Jan 29 15:22 ../src/kernels/gfx942.kdb.bz2
```
One will observe that the `GIT LFS files not pulled down, skipped` is **NOT** printed where they should have.

**[Debugging]:**:
If we make the following changes:
```
# Begin KDB package creation
set_property(GLOBAL PROPERTY LFS_MISSING_FILES)
function(install_kdb FILE_NAME COMPONENT_NAME)
    file(READ ${FILE_NAME} __contents LIMIT 7)
    get_filename_component(__fname ${FILE_NAME} NAME_WLE)
    if(__contents MATCHES "version")
        message("****DEBUGMARK1 ${LFS_MISSING_FILES}")
        set(LFS_MISSING_FILES ${LFS_MISSINGFILES} ${__fname} PARENT_SCOPE)
        message("****DEBUGMARK2 ${LFS_MISSING_FILES}")
    else()
```
We will observe:
```
****DEBUGMARK1 
****DEBUGMARK2 
****DEBUGMARK1 gfx90a.kdb
****DEBUGMARK2 gfx90a.kdb
****DEBUGMARK1 gfx1030.kdb
****DEBUGMARK2 gfx1030.kdb
****DEBUGMARK1 gfx908.kdb
****DEBUGMARK2 gfx908.kdb
****DEBUGMARK1 gfx906.kdb
****DEBUGMARK2 gfx906.kdb
CMake Warning at CMakeLists.txt:551 (message):
  GIT LFS files not pulled down, skipped: gfx900.kdb
  
```
@apwojcik you can see in the above function, `LFS_MISSING_FILES` unfortunately gets updated every time it is called instead of accumulating the list.

**Comment by apwojcik** [comment]
Fixed in #2710
